main.go

```diff
@@ -42,9 +42,33 @@ var (
 	fUpgrade  = flag.Bool("u", false, "check for the latest tagged version of main packages")
 	fNoNet    = flag.Bool("nonet", false, "prevent network access")
 	fDebug    = flag.Bool("debug", false, "print debug information")
-	fTags     = flag.String("tags", "", "build tags to apply; go help build for more information")
+
+	fTags      *string
+	envGOFLAGS string
 )
 
+func init() {
+	// Set the default value of the -tags flag to be value specified via GOFLAGS; also
+	// clean the GOFLAGS env var for any subsequent go commands we run in the process.
+	var defaultTags string
+	var goenvVals []string
+	if gf := os.Getenv("GOFLAGS"); gf != "" {
+		var tagsVal *string
+		for _, v := range strings.Fields(gf) {
+			if strings.HasPrefix(v, "-tags=") {
+				tagsVal = &v
```

Comment:
This is triggering alarm bells, because we're taking the address of a loop variable that will change further down  the loop. I don't really understand what's going on with envGOFLAGS here (why do you need it? why split then join again?). Perhaps a comment might help?

Reply:
Thanks; Monday morning...